user.go

```diff
@@ -193,14 +229,20 @@ func (c *Client) CurrentUserFollows(t string, ids ...ID) ([]bool, error) {
 	return result, nil
 }
 
-func (c *Client) modifyFollowers(usertype string, follow bool, ids ...ID) error {
+const (
+	followingResource = "following"
+	albumsResource    = "albums"
+	tracksResource    = "tracks"
+)
+
+func (c *Client) modifyResource(resourceType string, usertype string, follow bool, ids ...ID) error {
 	if l := len(ids); l == 0 || l > 50 {
 		return errors.New("spotify: Follow/Unfollow supports 1 to 50 IDs")
 	}
 	v := url.Values{}
 	v.Add("type", usertype)
```

Comment:
It looks like the `type` parameter is only necessary for updating followers.

In addition, the error we issue if too many IDs are provided is specific to follow/unfollow.

Its starting to seem like we should have two separate methods - one for modifying followers, and one generic one for library manipulation (saving tracks, albums, shows, etc.)